wrlinux/wrlinux.go

```diff
@@ -0,0 +1,273 @@
+		}
+
+		if err = utils.SaveCVEPerYear(filepath.Join(utils.VulnListDir(), windriverDir), vuln.Candidate, vuln); err != nil {
+			return xerrors.Errorf("error in save: %w", err)
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return xerrors.Errorf("error in walk: %w", err)
+	}
+	return nil
+}
+
+func parse(r io.Reader) (vuln *Vulnerability, err error) {
+	vuln = &Vulnerability{}
+	vuln.Patches = map[Package]Statuses{}
+
+	all, err := ioutil.ReadAll(r)
+	if err != nil {
+		return nil, err
+	}
+	lines := strings.Split(string(all), "\n")
+
+	for i := 0; i < len(lines); i++ {
+		line := lines[i]
+
+		// Skip
+		if strings.HasPrefix(line, "#") || line == "" {
+			continue
+		}
+
+		// Parse Candidate
+		if strings.HasPrefix(line, "Candidate:") {
+			line = strings.TrimPrefix(line, "Candidate:")
+			vuln.Candidate = strings.TrimSpace(line)
+			continue
+		}
+
+		// Parse PublicDate
+		if strings.HasPrefix(line, "PublicDate:") {
+			line = strings.TrimPrefix(line, "PublicDate:")
+			line = strings.TrimSpace(line)
+			vuln.PublicDate, _ = dateparse.ParseAny(line)
+			continue
+		}
+
+		// Parse Description
+		if strings.HasPrefix(line, "Description:") {
+			var description []string
+			for strings.HasPrefix(lines[i+1], " ") {
+				i++
+				line = strings.TrimSpace(lines[i])
+				description = append(description, line)
+			}
+			vuln.Description = strings.Join(description, " ")
+			continue
+		}
+
+		// Parse References
+		if strings.HasPrefix(line, "References:") {
+			for strings.HasPrefix(lines[i+1], " ") {
+				i++
+				line = strings.TrimSpace(lines[i])
+				vuln.References = append(vuln.References, line)
+			}
+			continue
+		}
+
+		// Parse Notes
+		if strings.HasPrefix(line, "Notes:") {
+			for strings.HasPrefix(lines[i+1], " ") {
+				i++
+				line = strings.TrimSpace(lines[i])
+				note := []string{line}
+				for strings.HasPrefix(lines[i+1], "  ") {
+					i++
+					l := strings.TrimSpace(lines[i])
+					note = append(note, l)
+				}
+				vuln.Notes = append(vuln.Notes, strings.Join(note, " "))
+			}
+			continue
+		}
+
+		// Parse Priority
+		if strings.HasPrefix(line, "Priority:") {
+			line = strings.TrimPrefix(line, "Priority:")
+			vuln.Priority = strings.TrimSpace(line)
+			continue
+		}
+
+		// Parse Bugs
+		if strings.HasPrefix(line, "Bugs:") {
+			for strings.HasPrefix(lines[i+1], " ") {
+				i++
+				line = strings.TrimSpace(lines[i])
+				vuln.Bugs = append(vuln.Bugs, line)
+			}
+			continue
+		}
+
+		// Parse Patches
+		// e.g. trusty/esm_vnc4: needs-triage
+		s := strings.SplitN(line, ":", 2)
+		if len(s) < 2 {
+			continue
+		}
+
+		status := strings.TrimSpace(s[1])
+
+		// Some advisories have status with "Patches_" prefix and it should be skipped
+		// e.g. Patches_qtwebkit-opensource-src: needs-triage
+		if isPatch(status) && !strings.HasPrefix(s[0], "Patches_") {
+			pkgRel := strings.SplitN(s[0], "_", 2)
+			release := Release(pkgRel[0])
+			pkgName := Package(strings.Trim(pkgRel[1], ":"))
```

Comment:
looks like we don't need trim `:`, because we did this here:
https://github.com/aquasecurity/vuln-list-update/blob/0b161fc3cdeb4b02c0296420e13a11818ed986c9/wrlinux/wrlinux.go#L230

Reply:
correct